clients/dotnet-core-rest/src/Tika.RestClient/ClientOptions.cs

```diff
@@ -1,16 +1,51 @@
+using System;
 using Microsoft.Extensions.Configuration;
 
 namespace Tika.RestClient
 {
     public class ClientOptions
     {
         public string TIKA_API_ENDPOINT { get; set; }
+        public bool TIKA_ENABLE_MULTI_CLUSTER { get; set; }
+        public string TIKA_MULTI_CLUSTER_HOSTNAME_PREFIX { get; set; }
         
         public ClientOptions() {}
 
         public ClientOptions(IConfiguration conf)
         {
-            TIKA_API_ENDPOINT = conf["TIKA_API_ENDPOINT"];
+            // Main config
+            // Will be ignored if TIKA_ENABLE_MULTI_CLUSTER is enabled
+            TIKA_API_ENDPOINT = ConfToString(conf, "TIKA_API_ENDPOINT", null);
+            
+            // Enable features
+            TIKA_ENABLE_MULTI_CLUSTER = ConfToBool(conf, "TIKA_ENABLE_MULTI_CLUSTER", false);
+            
+            // Multi-cluster config
+            TIKA_MULTI_CLUSTER_HOSTNAME_PREFIX = ConfToString(conf, "TIKA_MULTI_CLUSTER_HOSTNAME_PREFIX", "http://tika-");
```

Comment:
Use options pattern for configs (IOptions) and ensure sensible defaults are set at the "property level"